### PR TITLE
add torchdata as prototype dependency

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

**Comment by pmeier** [comment]
Blocked by pytorch/data#20.

File: torchvision/prototype/datasets/__init__.py [hunk 1]
```diff
@@ -1,3 +1,13 @@
+try:
+    import torchdata
+except (ModuleNotFoundError, TypeError) as error:
```
Comment (pmeier) [inline torchvision/prototype/datasets/__init__.py:3]: The `torchdata` library hosted on PyPI fails for me at import with an `TypeError`. This is why we can aggregate both failure cases here.

File: torchvision/prototype/datasets/__init__.py [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+    raise ModuleNotFoundError(
+        "`torchvision.prototype.datasets` depends on PyTorch's `torchdata` (https://github.com/pytorch/data). "
+        "You can install it with `pip install git+https://github.com/pytorch/data.git`. "
```
Comment (fmassa) [inline torchvision/prototype/datasets/__init__.py:6]: Do you need to add `@main` in the uri ? You've added it in the other commands

File: torchvision/prototype/datasets/__init__.py (hunk 1 above)
Comment (fmassa) [reply torchvision/prototype/datasets/__init__.py:3]: Chatted with @pmeier , this is a bug in the `torchdata` from PyPI (which is unrelated to the `torchdata` we are depending on.

File: torchvision/prototype/datasets/__init__.py (hunk 2 above)
Comment (pmeier) [reply torchvision/prototype/datasets/__init__.py:6]: Nope, if you don't specify a branch / tag / ..., `pip` uses the default branch. I'll remove it in the other places to be consistent.

**Review by fmassa — APPROVED** [review]
Thanks! Only have one quick comment

File: torchvision/prototype/datasets/__init__.py (hunk 1 above)
Comment (pmeier) [reply torchvision/prototype/datasets/__init__.py:3]: If you do `pip install torchdata`, `python -c "import torchdata"` fails with 

```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

**Pull request merged**